ClinGen/clincoded
# Connects to #1420.  Publish classifications from the GCI to the ClinGen website.

To prepare a non-local instance to be able to publish classifications, the following commands need to be run (on the instance).

To install the required kafka client:
1. `wget -qO - https://packages.confluent.io/deb/4.1/archive.key | sudo apt-key add -`
2. `sudo add-apt-repository "deb [arch=amd64] https://packages.confluent.io/deb/4.1 stable main"`
3. `sudo apt-get update && sudo apt-get install librdkafka-dev python-dev`
4. `sudo apt install python3-pip`
5. `sudo /usr/bin/pip3 install confluent-kafka`

To add a needed variable to the environment:
1. `export KAFKA_CERT_PW=[password]`
2. `echo "export KAFKA_CERT_PW=[password]" >> ~/.bashrc`

To pull in a previous version of a file and restart the application:
1. `cd /srv/clincoded`
2. `sudo -u clincoded git checkout acb95eb480e608545cfd6dbe0dac6f86f6a04659 src/clincoded/__init__.py`
3. `sudo service apache2 restart`

Steps to test #1420:
1. After logging in and selecting an affiliation, go to a gene-disease record and click the "Classification Matrix" button.
2. Click the "Save" button to initiate the approval process.
3. Proceed through the approval process to see a "Publish Classification" panel and continue further to publish the message.
4. If the operation is successful (no error message), the page should refresh and some data related to the publish event should appear beneath the corresponding approved classification (within the same snapshot view) and in the header.

Other items to test:
1. From an approved, but not yet published, classification, click the "Publish Summary" button to publish the classification (should bring up the aforementioned panel).
2. When publishing, if an archived (older) classification is currently published, that archived classification should be automatically unpublished.
3. From a published classification, click the "Unpublish Summary" button to unpublish the classification (panel appears slightly different, but behaves like the publish version).
4. For classifications created by a user in the same affiliation, the dashboard should indicate which have been published.
